### PR TITLE
Deal with zero-weight edges

### PR DESCRIPTION
**Pull request by dourouc05**
Issue #24. 

Question: maybe it would be better to return false in this case?

### CODE REVIEW

File: src/simpleweighteddigraph.jl [hunk 1]
```diff
@@ -96,6 +96,11 @@ function add_edge!(g::SimpleWeightedDiGraph, e::SimpleWeightedGraphEdge)
     d = T(d_)
     (s in vertices(g) && d in vertices(g)) || return false
     g.weights[d, s] = w
+	
+    if w == zero(U)
+        warn("Note: adding edges with a zero weight to this graph type has no effect.", once=true)
```
Comment (sbromberger) [inline src/simpleweighteddigraph.jl]: Actually, it's more insidious: it has no effect if the edge doesn't already exist. If the edge exists, it actually replaces the existing weight with a structural zero. We should absolutely return `false` here, and we should put this check *before* we try to assign `w` to `g.weights`  (that is, make it the first check we do in the function.)

File: src/simpleweightedgraph.jl [hunk 2]
```diff
@@ -102,6 +102,11 @@ function add_edge!(g::SimpleWeightedGraph, e::SimpleWeightedGraphEdge)
     (s in vertices(g) && d in vertices(g)) || return false
     g.weights[d, s] = w
     g.weights[s, d] = w
+	
+    if w == zero(U)
+        warn("Note: adding edges with a zero weight to this graph type has no effect.", once=true)
+    end
+
```
Comment (sbromberger) [inline src/simpleweightedgraph.jl]: See above: move this to the beginning, and immediately return false.

**Review by sbromberger — COMMENTED** [review]
We'll also need some tests.

**Comment by dourouc05** [comment]
Here you are. I moved the test just after `w` exists. For the tests, I can only add it once (and not within the loop, for example), as it warns only once…

**Comment by sbromberger** [comment]
Finally got it to pass tests (I kept making silly mistakes). Thanks for your help!

**Comment by dourouc05** [comment]
Great, thanks!

**Pull request merged**